### PR TITLE
fix(www): adds basic server connection toggle timeouts

### PR DESCRIPTION
**Pull request by daniellacosse**
This stops the user from spamming the connect/disconnect button, potentially causing a race condition

### CODE REVIEW

File: src/www/app/app.ts [hunk 1]
```diff
@@ -408,6 +414,9 @@ export class App {
       throw new Error(`disconnectServer event had no server ID`);
     }
 
+    if (!this.canServerConnectionChange(serverId)) return;
```
Comment (jyyi1) [inline src/www/app/app.ts:417]: *minor* I feel we can move this `if` statement to `setServerConnectionChangeTimeout` to reduce the duplicated code here.

Comment (daniellacosse) [reply src/www/app/app.ts:417]: We'd also still need to handle the value returned from `setServerConnectionChangeTimeout` and return from this function anyway to short circuit it, so I'm not sure what that would DRY up, if I'm understanding you correctly.

Are you suggesting something like this?

```js
const didThrottle = this.throttleServerConnectionChange(serverId, DEFAULT_SERVER_CONNECTION_STATUS_CHANGE_TIMEOUT);


if (didThrottle) return;
```

Comment (daniellacosse) [reply src/www/app/app.ts:417]: Ah true, that could be come a single line - one moment

**Review by jyyi1 — APPROVED** [review]
LTGM, thx!

**Pull request merged**